lib/llvm-backend/src/code.rs

```diff
@@ -1486,8 +1516,14 @@ impl<'ctx> FunctionCodeGenerator<CodegenError> for LLVMFunctionCodeGenerator<'ct
                 } = &frame
                 {
                     if let IfElseState::If = if_else_state {
-                        builder.position_at_end(*if_else);
-                        builder.build_unconditional_branch(*next);
+                        if if_else.get_first_instruction().is_none() {
```

Comment:
Can you add a comment before this statement to give context?

```suggestion
                        // We remove unneeded basic blocks, in case is empty
                        if if_else.get_first_instruction().is_none() {
```